File: Reserva-Hotel-com-Front-main/ProjetoCs-Reserva_de_Hotel-main/ReservaDeHotel/Models/pagamento.cs
using System.ComponentModel.DataAnnotations;

namespace ReservaDeHotel.Models;
public class Pagamento
{
    [Key]
    public int? IdPagamento {get;set;}
    public int? ReservaId { get; set; }
    public ReservaHotel? Reserva { get; set; }
    public string? MetodoPagamento { get; set;}
    public decimal? Valor { get; set;} 
    public DateTime? DataPagamento { get; set;}
}

File: Reserva-Hotel-com-Front-main/ReservaHotelFront/src/app/Pagamento.ts
import { ReservaHotel } from './ReservaHotel';

export class Pagamento {
    idPagamento: number = 0;
    metodoPagamento: string = '';
    valor: number = 0;
    dataPagamento: Date = new Date();
    reservaHotelId: number = 0;
    reservaHotel: ReservaHotel | undefined;
}
